miner/worker.go

```diff
@@ -661,6 +669,17 @@ func (w *worker) resultLoop() {
 				logs = append(logs, receipt.Logs...)
 			}
 
+			if prev, ok := w.recentMinedBlocks.Get(block.NumberU64()); ok {
+				prevParent, _ := prev.(common.Hash)
+				if prevParent == block.ParentHash() {
+					log.Error("Reject Double Sign!!", "block", block.NumberU64(),
+						"hash", block.Hash(), "root", block.Root(),
+						"ParentHash", block.ParentHash(), "prevParent", prevParent)
+					continue
+				}
+			}
+			w.recentMinedBlocks.Add(block.NumberU64(), block.ParentHash())
```

Comment:
In most cases,  the item in the LRU cache never been used, so when we pop the item from cache , the order is  not certain.

Reply:
as noted in the comment